modelci/cli/__init__.py

```diff
@@ -11,20 +11,123 @@
+        self._annotation = annotation
+
+    @staticmethod
+    def _get_parser(type_):
+        if inspect.isclass(type_) and issubclass(type_, BaseModel):
+            return type_.parse_obj
+        else:
+            return type_
+
+    def convert(self, value, param, ctx):
+        key_type, value_type = self._annotation.__args__
+
+        try:
+            return {
+                self._get_parser(key_type)(k): self._get_parser(value_type)(v)
+                for k, v in ast.literal_eval(str(value)).items()
+            }
+        except ValidationError as exc:
+            typer.echo(exc, err=True, color=True)
+            raise typer.Exit(422)
+
+
+def _get_click_type_wrapper(get_click_type):
+    def supersede_get_click_type(
+            *, annotation: Any, parameter_info: typer.main.ParameterInfo
+    ) -> click.ParamType:
+        """
+        Monkey-patched typer for the support of customer type such as `pydantic` models and `Dict`.
+
+        Originated from https://github.com/tiangolo/typer/issues/111#issuecomment-635512182.
+        """
+
+        if isgeneric(annotation) and annotation.__origin__ is dict:
+            return DictParamType(annotation)
+        elif lenient_issubclass(annotation, BaseModel):
+            return PydanticModelParamType(annotation)
+        elif lenient_issubclass(annotation, NamedEnum):
+            return click.Choice(
+                [item.name for item in annotation],
+                case_sensitive=parameter_info.case_sensitive,
+            )
+        else:
+            return get_click_type(annotation=annotation, parameter_info=parameter_info)
 
+    return supersede_get_click_type
+
+
+def _generate_enum_convertor_wrapper(enum_converter_factory):
+    """Wrapper for generate enum converter."""
+
+    def generate_named_enum_converter(enum):
+        def convertor(value: Any) -> Any:
+            if value is not None:
+                low = str(value).lower()
+                if low in lower_val_map:
+                    key = lower_val_map[low]
+                    return enum(key)
+
+        if issubclass(enum, NamedEnum):
+            lower_val_map = {str(val.name).lower(): val for val in enum}
+            return convertor
+        else:
+            return enum_converter_factory
+
+    return generate_named_enum_converter
+
+
+typer.main.get_click_type = _get_click_type_wrapper(typer.main.get_click_type)
+typer.main.generate_enum_convertor = _generate_enum_convertor_wrapper(typer.main.generate_enum_convertor)
+app = typer.Typer()
+
+
+@app.callback()
+def callback():
+    """
+    A complete MLOps platform for managing, converting and profiling models and
+    then deploying models as cloud services (MLaaS)
+    """
+
+
+app.add_typer(modelhub_app, name='modelhub_typer')
+typer_click_object: click.Group = typer.main.get_command(app)  # noqa
+typer_click_object.add_command(service)
+typer_click_object.add_command(modelhub)
+
+if __name__ == '__main__':
```

Comment:
please remove main function here, __init__.py is not the program gateway

Reply:
This is required to start the CLI.